### PR TITLE
Add ExpressionEngine 3 CSM support

### PR DESCRIPTION
**Pull request by jjpmann**

### CODE REVIEW

File: src/Composer/Installers/ExpressionEngineInstaller.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+namespace Composer\Installers;
+
+use Composer\Package\PackageInterface;
+
+class ExpressionEngineInstaller extends BaseInstaller
+{
+
+    protected $locations = [];
```
Comment (niksamokhvalov) [inline src/Composer/Installers/ExpressionEngineInstaller.php]: PHP 5.3 not supported short tags.

**Comment by niksamokhvalov** [comment]
Hi, write test, please. Example:
https://github.com/composer/installers/blob/master/tests/Composer/Installers/Test/InstallerTest.php#L87
https://github.com/composer/installers/blob/master/tests/Composer/Installers/Test/InstallerTest.php#L205

**Comment by jjpmann** [comment]
Please let me know if anything else is needed for this and Thanks.

File: tests/Composer/Installers/Test/InstallerTest.php [hunk 2]
```diff
@@ -115,6 +115,10 @@ public function dataForTestSupport()
             array('dokuwiki-template', true),
             array('drupal-module', true),
             array('dolibarr-module', true),
+            array('ee3-theme', true),
+            array('ee3-addon', true),
+            array('ee2-theme', true),
+            array('ee3-addon', true),
```
Comment (niksamokhvalov) [inline tests/Composer/Installers/Test/InstallerTest.php]: May be `ee2`?

**Comment by slbmeh** [comment]
AFAIK ee2 addons can work for ee3.  Shouldn't it be just be `ee-addon` and `ee-theme` instead?

The paths are different between the two, but that's configurable in expression engine and composer installers both already.

**Comment by jjpmann** [comment]
ee3 introduced a whole new UI and other breaking changes, ee2 add-ons
require updates to work

**Comment by jjpmann** [comment]
Hi @niksamokhvalov, Is there anything else needed to get this added? Please let me know.

**Pull request merged**